pkg/gateway/path/resolver.go

```diff
@@ -15,12 +15,12 @@ const (
 	Separator = "/"
 
 	rePath      = "(?P<path>.*)"
-	reReference = `(?P<ref>\w[-\w]*)`
+	reReference = `(?P<ref>\w[-\w]*[~^0-9]*)`
 )
 
 var (
 	EncodedPathRe          = regexp.MustCompile(fmt.Sprintf("^/?%s/%s", reReference, rePath))
-	EncodedPathReferenceRe = regexp.MustCompile(fmt.Sprintf("^/?%s", reReference))
+	EncodedPathReferenceRe = regexp.MustCompile(fmt.Sprintf("^/?%s$", reReference))
```

Comment:
To be clear: I am asking why we cannot just write
```suggestion
	EncodedPathReferenceRe = regexp.MustCompile("^/?[^/]?*$")
```
and be done with it: RevParse will do the rest.

Reply:
Cool, with the new `reReference` I understand what this does!  BTW, I am not sure that we need `$` here, I'm pretty sure it doesn't _change_ any behaviour for a legal path... but it *is* easier to understand what it does.